### ISSUE TITLE
Unable to retrieve child resources.

### ISSUE THREAD

**Issue by JCKodel** [microsoft/AzureStorageExplorer#3968]
Unable to retrieve child resources.

Details:
{
  "name": "Error",
  "message": "Hub Controller process exited prematurely with signal 'SIGABRT'.",
  "__zone_symbol__error": {
    "originalStack": "Error: Hub Controller process exited prematurely with signal 'SIGABRT'.\n    at AppInsightsAsyncCorrelatedErrorWrapper.ZoneAwareError (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/zone.js/dist/zone-node.js:811:33)\n    at new AppInsightsAsyncCorrelatedErrorWrapper (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/applicationinsights/out/AutoCollection/CorrelationContextManager.js:193:18)\n    at ControllerConnection.onControllerProcessExited (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/microsoft-servicehub/controllerConnection.js:135:21)\n    at ChildProcess.<anonymous> (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/microsoft-servicehub/controllerConnection.js:96:93)\n    at Object.onceWrapper (events.js:300:26)\n    at ZoneDelegate.invokeTask (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/zone.js/dist/zone-node.js:275:35)\n    at Zone.runTask (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/zone.js/dist/zone-node.js:151:47)\n    at ChildProcess.ZoneTask.invoke (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/zone.js/dist/zone-node.js:345:33)\n    at ChildProcess.emit (events.js:215:7)\n    at Process.ChildProcess._handle.onexit (internal/child_process.js:272:12)",
    "zoneAwareStack": "Error: Hub Controller process exited prematurely with signal 'SIGABRT'.\n    at AppInsightsAsyncCorrelatedErrorWrapper.ZoneAwareError (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/zone.js/dist/zone-node.js:811:33)\n    at new AppInsightsAsyncCorrelatedErrorWrapper (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/applicationinsights/out/AutoCollection/CorrelationContextManager.js:193:18)\n    at ControllerConnection.onControllerProcessExited (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/microsoft-servicehub/controllerConnection.js:135:21)\n    at ChildProcess.<anonymous> (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/microsoft-servicehub/controllerConnection.js:96:93)\n    at Object.onceWrapper (events.js:300:26)\n    at ZoneDelegate.invokeTask (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/zone.js/dist/zone-node.js:275:35)\n    at Zone.runTask (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/zone.js/dist/zone-node.js:151:47)\n    at ChildProcess.ZoneTask.invoke (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/zone.js/dist/zone-node.js:345:33)\n    at ChildProcess.emit (events.js:215:7)\n    at Process.ChildProcess._handle.onexit (internal/child_process.js:272:12)"
  },
  "stack": "Error: Hub Controller process exited prematurely with signal 'SIGABRT'.\n    at AppInsightsAsyncCorrelatedErrorWrapper.ZoneAwareError (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/zone.js/dist/zone-node.js:811:33)\n    at ControllerConnection.onControllerProcessExited (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/microsoft-servicehub/controllerConnection.js:135:21)\n    at ChildProcess.<anonymous> (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/microsoft-servicehub/controllerConnection.js:96:93)\n    at Object.onceWrapper (events.js:300:26)\n    at ZoneDelegate.invokeTask (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/zone.js/dist/zone-node.js:275:35)\n    at Zone.runTask (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/zone.js/dist/zone-node.js:151:47)\n    at ChildProcess.ZoneTask.invoke (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/zone.js/dist/zone-node.js:345:33)\n    at ChildProcess.emit (events.js:215:7)\n    at Process.ChildProcess._handle.onexit (internal/child_process.js:272:12)\nFrom previous event:\n    at AppInsightsAsyncCorrelatedErrorWrapper.ZoneAwareError (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/zone.js/dist/zone-node.js:811:33)\n    at defer (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/microsoft-servicehub/node_modules/q/q.js:582:19)\n    at Promise.then (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/microsoft-servicehub/node_modules/q/q.js:828:20)\n    at ControllerConnection.connectImpl (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/microsoft-servicehub/controllerConnection.js:48:14)\n    at Function.ControllerConnection.connect (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/microsoft-servicehub/controllerConnection.js:31:27)\n    at HubLookup.getLocationService (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/microsoft-servicehub/hubLookup.js:20:80)\n    at HubClient.requestServiceWithRetries (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/microsoft-servicehub/hubClient.js:74:21)\n    at HubClient.requestService (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/microsoft-servicehub/hubClient.js:62:21)\n    at ElectronAadAccountProviderClient.<anonymous> (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/identity-client/dist/AadAccountProviderClient.js:57:74)\n    at Generator.next (<anonymous>)\n    at /Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/identity-client/dist/AadAccountProviderClient.js:10:71\n    at new ZoneAwarePromise (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/zone.js/dist/zone-node.js:551:29)\n    at __awaiter (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/identity-client/dist/AadAccountProviderClient.js:6:12)\n    at /Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/identity-client/dist/AadAccountProviderClient.js:56:27",
  "originalStack": "Error: Hub Controller process exited prematurely with signal 'SIGABRT'.\n    at AppInsightsAsyncCorrelatedErrorWrapper.ZoneAwareError (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/zone.js/dist/zone-node.js:811:33)\n    at new AppInsightsAsyncCorrelatedErrorWrapper (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/applicationinsights/out/AutoCollection/CorrelationContextManager.js:193:18)\n    at ControllerConnection.onControllerProcessExited (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/microsoft-servicehub/controllerConnection.js:135:21)\n    at ChildProcess.<anonymous> (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/microsoft-servicehub/controllerConnection.js:96:93)\n    at Object.onceWrapper (events.js:300:26)\n    at ZoneDelegate.invokeTask (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/zone.js/dist/zone-node.js:275:35)\n    at Zone.runTask (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/zone.js/dist/zone-node.js:151:47)\n    at ChildProcess.ZoneTask.invoke (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/zone.js/dist/zone-node.js:345:33)\n    at ChildProcess.emit (events.js:215:7)\n    at Process.ChildProcess._handle.onexit (internal/child_process.js:272:12)",
  "zoneAwareStack": "Error: Hub Controller process exited prematurely with signal 'SIGABRT'.\n    at AppInsightsAsyncCorrelatedErrorWrapper.ZoneAwareError (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/zone.js/dist/zone-node.js:811:33)\n    at new AppInsightsAsyncCorrelatedErrorWrapper (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/applicationinsights/out/AutoCollection/CorrelationContextManager.js:193:18)\n    at ControllerConnection.onControllerProcessExited (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/microsoft-servicehub/controllerConnection.js:135:21)\n    at ChildProcess.<anonymous> (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/microsoft-servicehub/controllerConnection.js:96:93)\n    at Object.onceWrapper (events.js:300:26)\n    at ZoneDelegate.invokeTask (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/zone.js/dist/zone-node.js:275:35)\n    at Zone.runTask (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/zone.js/dist/zone-node.js:151:47)\n    at ChildProcess.ZoneTask.invoke (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/zone.js/dist/zone-node.js:345:33)\n    at ChildProcess.emit (events.js:215:7)\n    at Process.ChildProcess._handle.onexit (internal/child_process.js:272:12)",
  "__zone_symbol__stack": "Error: Hub Controller process exited prematurely with signal 'SIGABRT'.\n    at AppInsightsAsyncCorrelatedErrorWrapper.ZoneAwareError (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/zone.js/dist/zone-node.js:811:33)\n    at ControllerConnection.onControllerProcessExited (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/microsoft-servicehub/controllerConnection.js:135:21)\n    at ChildProcess.<anonymous> (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/microsoft-servicehub/controllerConnection.js:96:93)\n    at Object.onceWrapper (events.js:300:26)\n    at ZoneDelegate.invokeTask (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/zone.js/dist/zone-node.js:275:35)\n    at Zone.runTask (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/zone.js/dist/zone-node.js:151:47)\n    at ChildProcess.ZoneTask.invoke (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/zone.js/dist/zone-node.js:345:33)\n    at ChildProcess.emit (events.js:215:7)\n    at Process.ChildProcess._handle.onexit (internal/child_process.js:272:12)\nFrom previous event:\n    at AppInsightsAsyncCorrelatedErrorWrapper.ZoneAwareError (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/zone.js/dist/zone-node.js:811:33)\n    at defer (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/microsoft-servicehub/node_modules/q/q.js:582:19)\n    at Promise.then (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/microsoft-servicehub/node_modules/q/q.js:828:20)\n    at ControllerConnection.connectImpl (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/microsoft-servicehub/controllerConnection.js:48:14)\n    at Function.ControllerConnection.connect (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/microsoft-servicehub/controllerConnection.js:31:27)\n    at HubLookup.getLocationService (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/microsoft-servicehub/hubLookup.js:20:80)\n    at HubClient.requestServiceWithRetries (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/microsoft-servicehub/hubClient.js:74:21)\n    at HubClient.requestService (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/microsoft-servicehub/hubClient.js:62:21)\n    at ElectronAadAccountProviderClient.<anonymous> (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/identity-client/dist/AadAccountProviderClient.js:57:74)\n    at Generator.next (<anonymous>)\n    at /Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/identity-client/dist/AadAccountProviderClient.js:10:71\n    at new ZoneAwarePromise (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/zone.js/dist/zone-node.js:551:29)\n    at __awaiter (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/identity-client/dist/AadAccountProviderClient.js:6:12)\n    at /Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/identity-client/dist/AadAccountProviderClient.js:56:27",
  "__zone_symbol__name": "Error",
  "__zone_symbol__message": "Hub Controller process exited prematurely with signal 'SIGABRT'.",
  "__zone_symbol__originalStack": "Error: Hub Controller process exited prematurely with signal 'SIGABRT'.\n    at AppInsightsAsyncCorrelatedErrorWrapper.ZoneAwareError (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/zone.js/dist/zone-node.js:811:33)\n    at new AppInsightsAsyncCorrelatedErrorWrapper (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/applicationinsights/out/AutoCollection/CorrelationContextManager.js:193:18)\n    at ControllerConnection.onControllerProcessExited (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/microsoft-servicehub/controllerConnection.js:135:21)\n    at ChildProcess.<anonymous> (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/microsoft-servicehub/controllerConnection.js:96:93)\n    at Object.onceWrapper (events.js:300:26)\n    at ZoneDelegate.invokeTask (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/zone.js/dist/zone-node.js:275:35)\n    at Zone.runTask (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/zone.js/dist/zone-node.js:151:47)\n    at ChildProcess.ZoneTask.invoke (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/zone.js/dist/zone-node.js:345:33)\n    at ChildProcess.emit (events.js:215:7)\n    at Process.ChildProcess._handle.onexit (internal/child_process.js:272:12)",
  "__zone_symbol__zoneAwareStack": "Error: Hub Controller process exited prematurely with signal 'SIGABRT'.\n    at AppInsightsAsyncCorrelatedErrorWrapper.ZoneAwareError (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/zone.js/dist/zone-node.js:811:33)\n    at new AppInsightsAsyncCorrelatedErrorWrapper (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/applicationinsights/out/AutoCollection/CorrelationContextManager.js:193:18)\n    at ControllerConnection.onControllerProcessExited (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/microsoft-servicehub/controllerConnection.js:135:21)\n    at ChildProcess.<anonymous> (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/microsoft-servicehub/controllerConnection.js:96:93)\n    at Object.onceWrapper (events.js:300:26)\n    at ZoneDelegate.invokeTask (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/zone.js/dist/zone-node.js:275:35)\n    at Zone.runTask (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/zone.js/dist/zone-node.js:151:47)\n    at ChildProcess.ZoneTask.invoke (/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/node_modules/zone.js/dist/zone-node.js:345:33)\n    at ChildProcess.emit (events.js:215:7)\n    at Process.ChildProcess._handle.onexit (internal/child_process.js:272:12)",
  "signal": "SIGABRT"
}

**Comment by MRayermannMSFT** [comment]
1. In Storage Explorer settings, turn on Verbose Authentication Logging under, Application -> Sign-in
2. Close Storage Explorer
3. Delete the contents of C:/<username>/AppData/Local/temp/servicehub/logs/
4. Start Storage Explorer
5. Do everything you do to reproduce this problem

After it reproduces, zip up C:/<username>/AppData/Local/temp/servicehub/logs/, and attach it here
Thanks.

**Comment by JCKodel** [comment]
@MRayermannMSFT After logging in 3 times (because Microsoft Authentication is a huge bugged mess):

```
Request Id: 199f5d6a-6900-40f6-bf38-6143ed380400
Correlation Id: 2adcae62-8367-48e6-bd6a-dd281e5513c1
Timestamp: 2021-02-08T20:53:53Z
Message: AADSTS90072: User account 'jckodel@xxx.com' from identity provider 'live.com' does not exist in tenant 'ESX' and cannot access the application '04b07795-8ddb-461a-bbee-02f9e1bf7b46'(Microsoft Azure CLI) in that tenant. The account needs to be added as an external user in the tenant first. Sign out and sign in again with a different Azure Active Directory user account
```

When I try to login with my Microsoft Account, it redirects me to GitHub, then it redirects me to a company I no longer work for more than 1 year, then it redirects me again to GitHub, then it fails with the above message (this was logged as an error 400 in the log), but the authentication succeeds anyway (despite the login window say it doesn't).

[logs.zip](https://github.com/microsoft/AzureStorageExplorer/files/5946794/logs.zip)

**Comment by MRayermannMSFT** [comment]
Can you try upgrading to at least version 1.19.0, try signing in, and then we can troubleshooting if you're still having issues with that version?

**Comment by JCKodel** [comment]
> Can you try upgrading to at least version 1.19.0, try signing in, and then we can troubleshooting if you're still having issues with that version?

Signing in? That's almost impossible in Storage Explorer.

I can do only if I set Device Code Flow (deprecated), otherwise, it redirects me to GitHub, then to MS again, then to GitHub again, then to some fucking company I don't work for years.

MS authentication is a fuckin mess.

**Comment by jinglouMSFT** [comment]
v1.19.0 uses a different authentication library which we hope to give you a better sign-in experience. Please consider giving it a try and let us know your experience. We'll be happy to help you out if this new library still doesn't address your situation. We understand your frustration and appreciate your patience.  We love to hear our customer feedback and we work hard to address the issues as quickly as we could.  We'd also appreciate it when you report issues or communicate with us if you could consider using only appropriate language.  This is a professional form.  There is no place for profanity.  Thanks.

**Comment by MRayermannMSFT** [comment]
> otherwise, it redirects me to GitHub, then to MS again, then to GitHub again, then to some fucking company I don't work for years.

That issue is something that we have addressed in version 1.19. There were many other people with that issue(#460) and for all of them the update has made using Storage Explore much easier.

**Comment by JCKodel** [comment]
> > otherwise, it redirects me to GitHub, then to MS again, then to GitHub again, then to some fucking company I don't work for years.
> 
> That issue is something that we have addressed in version 1.19. There were many other people with that issue(#460) and for all of them the update has made using Storage Explore much easier.

As I said, now it is impossible to authenticate, except using the deprecated authentication mode. When that is removed, I will just be forced to use only web Azure portal. So, no, the fix won't fix a thing!
